code/go/example.go

```diff
@@ -1,68 +1,49 @@
-// go mod init main
+// go mod init my-module-path
 // go run example.go
 package main
 
 import (
 	"fmt"
 	"github.com/neo4j/neo4j-go-driver/v4/neo4j"
-	"io"
-	"reflect"
 )
 
 func main() {
-	results, err := runQuery("neo4j+s://demo.neo4jlabs.com:7687", "movies", "mUser", "s3cr3t")
+	credentials := neo4j.BasicAuth("mUser", "s3cr3t", "")
+	driver, err := neo4j.NewDriver("neo4j+s://demo.neo4jlabs.com:7687", credentials)
 	if err != nil {
-		panic(err)
+		panic("Could not create driver")
 	}
-	for _, result := range results {
-		fmt.Println(result)
+	defer driver.Close()
+	results, err := runQuery(driver, getDairyProducts)
+	if err != nil {
+		panic(err)
 	}
+	fmt.Println(results)
 }
 
-func runQuery(uri, database, username, password string) (result []string, err error) {
-	driver, err := neo4j.NewDriver(uri, neo4j.BasicAuth(username, password, ""))
+func runQuery(driver neo4j.Driver, txFunc neo4j.TransactionWork) (interface{}, error) {
+	session := driver.NewSession(neo4j.SessionConfig{DatabaseName: "movies"})
+	defer session.Close()
+	return session.ReadTransaction(txFunc)
+}
+
+func getDairyProducts(transaction neo4j.Transaction) (interface{}, error) {
+	cursor, err := transaction.Run(
+		`MATCH (m:Movie {title:$movieTitle})<-[:ACTED_IN]-(a:Person) 
+		RETURN a.name as actorName`,
+		map[string]interface{}{"movieTitle": "The Matrix"})
 	if err != nil {
 		return nil, err
 	}
-	defer func() {err = handleClose(driver, err)}()
-	session := driver.NewSession(neo4j.SessionConfig{AccessMode: neo4j.AccessModeRead, DatabaseName: database})
-	defer func() {err = handleClose(session, err)}()
-	results, err := session.ReadTransaction(func(transaction neo4j.Transaction) (interface{}, error) {
-		result, err := transaction.Run(
-			`
-			MATCH (m:Movie {title:$movieTitle})<-[:ACTED_IN]-(a:Person) RETURN a.name as actorName
-			`, map[string]interface{}{
-				"movieTitle": "The Matrix",
-			})
-		if err != nil {
-			return nil, err
-		}
-		var arr []string
-		for result.Next() {
-			value, found := result.Record().Get("actorName")
-			if found {
-				arr = append(arr, value.(string))
-			}
-		}
-		if err = result.Err(); err != nil {
-			return nil, err
+	var results []string
+	for cursor.Next() {
+		value, found := cursor.Record().Get("actorName")
+		if found {
+			results = append(results, value.(string))
 		}
-		return arr, nil
-	})
-	if err != nil {
-		return nil, err
 	}
-	result = results.([]string)
-	return result, err
-}
-
-func handleClose(closer io.Closer, previousError error) error {
```

Comment:
note: I purposefully removed this. 
It's the correct way to deal with deferred calls' errors but is super verbose and maybe not much of interest for a homepage example